bin/arnold

```diff
@@ -0,0 +1,1009 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# Docker
+declare -r ARNOLD_IMAGE_NAME="${ARNOLD_IMAGE_NAME:-fundocker/arnold}"
+declare -r ARNOLD_IMAGE_TAG="${ARNOLD_IMAGE_TAG:-5.22.0}"
+declare -r ARNOLD_IMAGE="${ARNOLD_IMAGE:-${ARNOLD_IMAGE_NAME}:${ARNOLD_IMAGE_TAG}}"
+declare -r ARNOLD_PROJECT_PATH="${PWD}"
+declare -r ARNOLD_TRAY_FILE="arnold.yml"
+declare -r DOCKER_USER="${DOCKER_USER:-$(id -u)}"
+
+# Arnold release info
+declare LATEST_RELEASE
+declare LATEST_VERSION
+
+# Script infos
+declare SCRIPT_NAME
+SCRIPT_NAME=$(basename "$0")
+declare SCRIPT_PATH
+SCRIPT_PATH="$(dirname "$(readlink -f "$0")")/${SCRIPT_NAME}"
+
+# The team file lists emails corresponding to public gpg keys that will be used
+# to encrypt ansible vault password using gpg.
+declare -r TEAM_FILE="${ARNOLD_PROJECT_PATH}/.team"
+
+# Set ansible vault password default to the corresponding environment variable
+# value (insecure mode, while not using gpg or --ask-vault-pass argument in
+# ansible-playbook command calls).
+declare ANSIBLE_VAULT_PASSWORD="${ANSIBLE_VAULT_PASSWORD}"
+
+# Terminal colors
+declare -r COLOR_BLUE='\033[0;36m'
+declare -r COLOR_DEFAULT='\033[0;39m'
+declare -r COLOR_GREEN='\033[0;32m'
+declare -r COLOR_ORANGE='\033[0;33m'
+declare -r COLOR_RED='\033[0;31m'
+declare -r COLOR_RESET='\033[0m'
+
+# Log levels
+declare -i LOG_LEVEL_CRITICAL=50
+declare -i LOG_LEVEL_ERROR=40
+declare -i LOG_LEVEL_WARNING=30
+declare -i LOG_LEVEL_SUCCESS=25
+declare -i LOG_LEVEL_INFO=20
+declare -i LOG_LEVEL_DEBUG=10
+declare -i LOG_LEVEL_NOTSET=0
+declare -i DEFAULT_LOG_LEVEL=${LOG_LEVEL_INFO}
+declare -i LOG_LEVEL=${DEFAULT_LOG_LEVEL}
+declare -A LOG_LEVELS=(
+  ["critical"]=${LOG_LEVEL_CRITICAL}
+  ["error"]=${LOG_LEVEL_ERROR}
+  ["warning"]=${LOG_LEVEL_WARNING}
+  ["success"]=${LOG_LEVEL_SUCCESS}
+  ["info"]=${LOG_LEVEL_INFO}
+  ["debug"]=${LOG_LEVEL_DEBUG}
+  ["notset"]=${LOG_LEVEL_NOTSET}
+)
+declare -A LOG_LEVEL_COLORS=(
+  ["critical"]=${COLOR_RED}
+  ["error"]=${COLOR_RED}
+  ["warning"]=${COLOR_ORANGE}
+  ["success"]=${COLOR_GREEN}
+  ["info"]=${COLOR_BLUE}
+  ["debug"]=${COLOR_DEFAULT}
+  ["notset"]=${COLOR_DEFAULT}
```

Comment:
`notset` ? Kesako ?

Reply:
I tried to mimic python logging module logic here. See: https://docs.python.org/3/howto/logging.html#logging-levels

Reply:
First time I see this... I'm not convinced it's a good thing but why not.